agr_literature_service/lit_processing/data_ingest/post_resource_to_db.py

```diff
@@ -49,9 +53,174 @@
+            new_entry[key] = entry[key]
+    return new_entry
+
+
+def process_resource_entry(db_session: Session, entry: Dict, xref_ref: Dict, ref_xref_valid: Dict) -> Tuple:
+    """Process json and add to db.
+    Adds resourses, cross references and editors.
+
+    :param db_session: database session
+    :param entry:      json format of dqm resource
+    :param xref_ref:   in the format xref_ref[prefix][identifier] = agr
+                       Used to ensure xrefs only connect to 1 resource.
+                       NOTE: Does not seem to get updated as resources are added.
+    :param ref_xref_valid: in the format ref_xref[agr][prefix]
     """
+    primary_id = entry['primaryId']
+    prefix, identifier, separator = split_identifier(primary_id)
+    if prefix in xref_ref:
+        if identifier in xref_ref[prefix]:
+            logger.info("%s\talready in", primary_id)
+            return True, ""
+
+    new_entry = remap_keys_get_new_entry(entry)
+    try:
+        resource_id = None
+        curie = get_next_resource_curie(db_session)
+
+        # cross_references and editors done seperately
+        # so do not want to pass them to the resource creator
+        # make a copy and deal with them after we have a resource_id
+        # to attach too.
+        cross_references = new_entry.get('cross_references', [])
+        if "cross_references" in new_entry:
+            del new_entry["cross_references"]
+        editors = new_entry.get('editors', [])
+        if "editors" in new_entry:
+            del new_entry["editors"]
+
+        new_entry['curie'] = curie
+        logger.info("Adding resource into database for '" + new_entry['iso_abbreviation'] + "'")
+        x = ResourceModel(**new_entry)
+        db_session.add(x)
+        db_session.flush()
+        db_session.commit()
+        db_session.refresh(x)
+        resource_id = x.resource_id
+
+        xref_okay, message = process_cross_references(db_session, resource_id, curie, cross_references, ref_xref_valid)
+
+        # Still process editors if xrefs fail?
+        process_editors(db_session, resource_id, editors)
+
+        if not xref_okay:
+            return xref_okay, message
+        return True, f"{primary_id}\t{curie}\n"
+    except Exception as e:
+        message = f"An error occurred when adding resource into database for '{entry}'. {e}\n"
+        logger.info(message)
+        db_session.rollback()
+        return False, message
+
+
+def post_resources(db_session: Session, input_path: str, input_mod: str, base_input_dir: str = base_path) -> None:      # noqa: C901
+    """
+    Parse the json file and load the data into the db after some remapping of keys etc.
+    i.e.
+    base_input_dir = "/user/bob/")
+    json_path = "sanitized_resources/"
+    post_resources(db, json_path, 'ZFIN', base_input_dir)
+
+    reads the file
+    /user/bob/sanitized_resource_json_updates/RESOURCE_ZFIN.json"
+
+    If base_input_dir not defined environment value of XML_BASE used.
```

Comment:
I think it's XML_PATH instead of XML_BASE, might be confusing to someone reading this later

Reply:
Sorted thanks.